src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -35,13 +35,28 @@ public static void ChangeExtension(string path, string newExtension, string expe
             Assert.Equal(expected, Path.ChangeExtension(path, newExtension));
         }
 
-        [Theory]
-        [InlineData("")]
-        [InlineData(" ")]
-        [InlineData("\r\n")]
-        public static void GetDirectoryName_EmptyOrWhitespace_Throws(string path)
+        [Fact]
+        public static void GetDirectoryName_EmptyThrows()
+        {
+            AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetDirectoryName(string.Empty));
+        }
+
+        [Theory,
+            InlineData(" "),
+            InlineData("\r\n")]
+        [ActiveIssue(21358)] // Pending CoreCLR behavior change
+        public static void GetDirectoryName_SpaceOrControlCharsThrowOnWindows(string path)
         {
-            AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetDirectoryName(path));
+            Action action = () => Path.GetDirectoryName(path);
+            if (PlatformDetection.IsWindows)
```

Comment:
This should be using PlatformSpecific attributes instead.

Also the test name has "Windows" in it but runs on Unix.